src/libostree/ostree-repo-pull.c

```diff
@@ -4172,6 +4152,42 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
                                  g_strdup (delta),
                                  csum_data);
           }
+
+        if (g_variant_lookup (additional_metadata, OSTREE_SUMMARY_MODE, "s", &remote_mode_str) &&
+            g_variant_lookup (additional_metadata, OSTREE_SUMMARY_TOMBSTONE_COMMITS, "b", &tombstone_commits))
+          {
+            if (!ostree_repo_mode_from_string (remote_mode_str, &pull_data->remote_mode, error))
+              goto out;
+            pull_data->has_tombstone_commits = tombstone_commits;
+          }
+        else if (pull_data->remote_repo_local == NULL)
```

Comment:
Hmm, don't we need to fallback also if the remote doesn't have a summary file? I.e. I think this needs to be moved out of the `if (pull_data->summary)` if-statement and executed if both it's a remote repo and we didn't already fetch the info we need from the optional summary file.